server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationMailet.java

```diff
@@ -0,0 +1,105 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.mailet;
+
+import java.time.ZonedDateTime;
+import java.util.concurrent.CompletableFuture;
+
+import javax.inject.Inject;
+import javax.mail.MessagingException;
+
+import org.apache.james.jmap.api.vacation.AccountId;
+import org.apache.james.jmap.api.vacation.Vacation;
+import org.apache.james.jmap.api.vacation.VacationRepository;
+import org.apache.james.jmap.utils.ZonedDateTimeProvider;
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailAddress;
+import org.apache.mailet.base.AutomaticallySentMailDetector;
+import org.apache.mailet.base.GenericMailet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class VacationMailet extends GenericMailet {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VacationMailet.class);
+
+    private final VacationRepository vacationRepository;
+    private final ZonedDateTimeProvider zonedDateTimeProvider;
+    private final AutomaticallySentMailDetector automaticallySentMailDetector;
+
+    @Inject
+    VacationMailet(VacationRepository vacationRepository, ZonedDateTimeProvider zonedDateTimeProvider, AutomaticallySentMailDetector automaticallySentMailDetector) {
+        this.vacationRepository = vacationRepository;
+        this.zonedDateTimeProvider = zonedDateTimeProvider;
+        this.automaticallySentMailDetector = automaticallySentMailDetector;
+    }
+
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        try {
+            ZonedDateTime processingDate = zonedDateTimeProvider.get();
+            mail.getRecipients()
+                .stream()
+                .map(mailAddress -> manageVacation(mailAddress, mail, processingDate))
+                .forEach(CompletableFuture::join);
+        } catch (Throwable e) {
+            LOGGER.warn("Can not process vacation for one or more recipients in {}", mail.getRecipients(), e);
+        }
+    }
+
+    public CompletableFuture<Void> manageVacation(MailAddress recipient, Mail processedMail, ZonedDateTime processingDate) {
+        AccountId accountId = AccountId.fromString(recipient.getLocalPart() + "@" + recipient.getDomain());
+        CompletableFuture<Vacation> vacationFuture = vacationRepository.retrieveVacation(accountId);
+        return vacationFuture.thenApply(vacation -> {
```

Comment:
Ok. I don't like this comment.

Could we please decide what syntax to choose, once and for all, when I need to execute some code upon completion of a future ?

I proposed two solutions : 
 - using completion stages. Completion stages clearly states we are doing side effects. Rejected by matthieu.
 - using thenApply. With side effects (whereas the type IMO suggest it looks like map). As I do not use it after this, but still need wait completion, hence, the Void type. Returning Void type is done with a ''return null;''.

This is my current proposal. Should I back up to completion stages ???